src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,145 @@
+    const { isFocusVisible, focusProps } = useFocusRing({ within: true, isTextInput: true });
+
+    const handleOnChange = (value: string) => {
+        onChange?.(value);
+    };
+
+    const handleCount = (value: string, type?: NumberInputIncrement) => {
+        let newValue = Number(value) || 0;
+        switch (type) {
+            case NumberInputIncrement.DECREMENT:
+                newValue -= 1;
+                break;
+            case NumberInputIncrement.INCREMENT:
+                newValue += 1;
+                break;
+            default:
+                newValue = Number(value);
+        }
+        if (inputElement.current) {
+            inputElement.current.value = newValue.toString();
+            handleOnChange(inputElement.current.value);
+        }
+    };
+
+    return (
+        <div data-test-id={`${dataTestId}-component`}>
+            <div
+                {...focusProps}
+                className={merge([
+                    'tw-flex tw-items-center tw-h-9 tw-gap-2 tw-px-3 tw-border tw-transition tw-rounded tw-text-s tw-font-sans tw-relative tw-bg-white dark:tw-bg-transparent',
+                    'tw-border-solid',
+                    'focus-within:tw-border-black-90 hover:tw-border-black-90',
+                    isFocusVisible && FOCUS_STYLE,
+                ])}
+            >
+                {!!decorator ? (
+                    <div
+                        className={merge([
+                            'tw-flex tw-items-center tw-justify-center tw-pl-1',
+                            disabled || readOnly ? 'tw-text-black-60' : 'tw-text-black-80',
+                        ])}
+                        data-test-id={`${dataTestId}-decorator`}
+                    >
+                        {decorator}
+                    </div>
+                ) : null}
+                <input
+                    {...props}
+                    id={useMemoizedId(propId)}
+                    ref={inputElement}
+                    name="fondue-number-input"
+                    type={numberInputType}
+                    className={merge([
+                        'tw-w-full tw-grow tw-border-none tw-outline-none tw-bg-transparent tw-hide-input-arrows',
+                        disabled || readOnly
+                            ? 'tw-text-black-40 tw-placeholder-black-30 dark:tw-text-black-30 dark:tw-placeholder-black-40'
+                            : 'tw-text-black tw-placeholder-black-60 dark:tw-text-white',
+                    ])}
+                    onClick={() => inputElement.current?.focus()}
+                    onChange={(event) => handleOnChange(event.currentTarget.value)}
+                    data-test-id={`${dataTestId}-input`}
+                    onBlur={onBlur}
+                    onKeyDown={onKeyDown}
+                    required={required}
+                    readOnly={readOnly}
+                    disabled={disabled}
+                    onFocus={(event) => (onFocus ? onFocus(event) : null)}
```

Comment:
```suggestion
                    onFocus={onFocus}
```
Not sure if types align tho

Reply:
`FocusEventHandle` is a generic.  Can you please clarify your thoughts on this?